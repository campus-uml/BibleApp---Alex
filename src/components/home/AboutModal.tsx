import { Button } from "@/components/ui/button";
import { Dialog, DialogContent, DialogTrigger } from "@/components/ui/dialog";
import { AboutMe } from "../AboutMe";
import { useEffect, useState } from "react";
import { DialogTitle } from "@radix-ui/react-dialog";

export const AboutModal = () => {
  const [isMobile, setIsMobile] = useState(false);

  useEffect(() => {
    const handleResize = () => {
      setIsMobile(window.innerWidth < 768);
    };

    handleResize();
    window.addEventListener("resize", handleResize);

    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);

  return (
    <Dialog>
      <DialogTrigger asChild>
 
        <Button
          variant="outline"
          className={`bg-slate-50 ${
            isMobile ? "text-slate-950 " : "text-black"
          } border-none`}
        >
          Acerca de
        </Button>
      </DialogTrigger>
      <DialogContent className="sm:max-w-[425px] bg-transparent border-none">
      <DialogTitle className="text-2xl font-semibold text-center text-white">
          Acerca del Desarrollador
        </DialogTitle>
        <AboutMe />
      </DialogContent>
    </Dialog>
  );
};
