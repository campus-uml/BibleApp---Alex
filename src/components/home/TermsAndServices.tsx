import { ScrollArea } from "@/components/ui/scroll-area";
import { Button } from "@/components/ui/button";

const TermsAndServices = () => {
  return (
    <div className="max-w-4xl mx-auto p-6 bg-background text-foreground">
      <h1 className="text-3xl font-bold mb-6 text-center">
        Términos y Servicios
      </h1>
      <ScrollArea className="h-[60vh] rounded-md border p-4">
        <div className="space-y-6">
          <section>
            <h2 className="text-2xl font-semibold mb-3">
              1. Aceptación de los Términos
            </h2>
            <p>
              Al utilizar nuestra aplicación bíblica, usted acepta estar sujeto
              a estos Términos y Servicios. Si no está de acuerdo con alguna
              parte de los términos, no podrá utilizar nuestra aplicación.
            </p>
          </section>

          <section>
            <h2 className="text-2xl font-semibold mb-3">
              2. Uso de la Aplicación
            </h2>
            <p>
              Nuestra aplicación está diseñada para proporcionar acceso a textos
              bíblicos y recursos relacionados. Usted se compromete a utilizar
              la aplicación solo para fines legales y de una manera que no
              infrinja los derechos de terceros.
            </p>
          </section>

          <section>
            <h2 className="text-2xl font-semibold mb-3">
              3. Propiedad Intelectual
            </h2>
            <p>
              Los textos bíblicos y otros contenidos proporcionados en la
              aplicación están sujetos a derechos de autor y otras leyes de
              propiedad intelectual. Se le concede un derecho limitado para uso
              personal y no comercial de estos contenidos.
            </p>
          </section>

          <section>
            <h2 className="text-2xl font-semibold mb-3">4. Privacidad</h2>
            <p>
              Su privacidad es importante para nosotros. Consulte nuestra
              Política de Privacidad para entender cómo recopilamos, usamos y
              protegemos sus datos personales.
            </p>
          </section>

          <section>
            <h2 className="text-2xl font-semibold mb-3">5. Modificaciones</h2>
            <p>
              Nos reservamos el derecho de modificar o reemplazar estos Términos
              en cualquier momento. Es su responsabilidad revisar estos Términos
              periódicamente para estar al tanto de cualquier cambio.
            </p>
          </section>

          <section>
            <h2 className="text-2xl font-semibold mb-3">
              6. Limitación de Responsabilidad
            </h2>
            <p>
              No seremos responsables de ningún daño indirecto, incidental,
              especial, consecuente o punitivo, o cualquier pérdida de
              beneficios o ingresos, ya sea incurrida directa o indirectamente,
              o cualquier pérdida de datos, uso, buena voluntad, u otras
              pérdidas intangibles, resultantes de su acceso o uso de la
              aplicación.
            </p>
          </section>

          <section>
            <h2 className="text-2xl font-semibold mb-3">7. Ley Aplicable</h2>
            <p>
              Estos Términos se regirán e interpretarán de acuerdo con las leyes
              de [su país/estado], sin tener en cuenta sus disposiciones sobre
              conflictos de leyes.
            </p>
          </section>
        </div>
      </ScrollArea>
      <div className="mt-6 flex justify-center">
        <Button className="px-6 py-2">Acepto los Términos y Servicios</Button>
      </div>
    </div>
  );
};

export default TermsAndServices;
