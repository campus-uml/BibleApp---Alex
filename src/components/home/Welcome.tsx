
import { Card, CardContent, CardDescription, CardFooter, CardHeader, CardTitle } from "@/components/ui/card";
import { Button } from "@/components/ui/button";
import { BookOpen, Search, Heart, BookMarked } from 'lucide-react';

const Welcome = () => {
  return (
    <div className="space-y-6">
      <Card className="w-full">
        <CardHeader>
          <CardTitle className="text-3xl font-bold text-center">Bienvenido a la Biblia Digital</CardTitle>
          <CardDescription className="text-center text-lg">
            Explora, estudia y comparte la palabra de Dios
          </CardDescription>
        </CardHeader>
        <CardContent className="text-center">
          <p className="text-muted-foreground">
            Selecciona un libro de la Biblia para comenzar a leer o utiliza la búsqueda para encontrar versículos específicos.
          </p>
        </CardContent>
        <CardFooter className="flex justify-center space-x-4">
          <Button variant="outline" className="flex items-center space-x-2">
            <BookOpen className="w-4 h-4" />
            <span>Leer</span>
          </Button>
          <Button variant="outline" className="flex items-center space-x-2">
            <Search className="w-4 h-4" />
            <span>Buscar</span>
          </Button>
        </CardFooter>
      </Card>

      <div className="grid gap-6 md:grid-cols-2 lg:grid-cols-3">
        <Card>
          <CardHeader>
            <CardTitle>Versículo del Día</CardTitle>
          </CardHeader>
          <CardContent>
            <p className="text-muted-foreground italic">
              "Porque de tal manera amó Dios al mundo, que ha dado a su Hijo unigénito, para que todo aquel que en él cree, no se pierda, mas tenga vida eterna."
            </p>
            <p className="mt-2 font-semibold">Juan 3:16</p>
          </CardContent>
          <CardFooter>
            <Button variant="ghost" className="w-full">
              <Heart className="w-4 h-4 mr-2" />
              Guardar
            </Button>
          </CardFooter>
        </Card>

        <Card>
          <CardHeader>
            <CardTitle>Lectura Recomendada</CardTitle>
          </CardHeader>
          <CardContent>
            <p className="text-muted-foreground">Explora el Libro de los Salmos, una colección de poemas y canciones que expresan una amplia gama de emociones humanas.</p>
          </CardContent>
          <CardFooter>
            <Button variant="ghost" className="w-full">
              <BookMarked className="w-4 h-4 mr-2" />
              Comenzar Lectura
            </Button>
          </CardFooter>
        </Card>

        <Card className="md:col-span-2 lg:col-span-1">
          <CardHeader>
            <CardTitle>Guía de Estudio</CardTitle>
          </CardHeader>
          <CardContent>
            <p className="text-muted-foreground">Descubre cómo realizar un estudio bíblico efectivo con nuestra guía paso a paso.</p>
          </CardContent>
          <CardFooter>
            <Button variant="ghost" className="w-full">Ver Guía</Button>
          </CardFooter>
        </Card>
      </div>
    </div>
  );
};

export default Welcome;
