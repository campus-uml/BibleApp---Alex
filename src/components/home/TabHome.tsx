"use client"

import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs"
import Welcome from "./Welcome"
import BibleVerse from "../BibleVerse"
import { useBible } from "@/context/BIbleContext"
import { useEffect, useState } from "react"
import { Heart, Share2 } from "lucide-react"
import { Button } from "@/components/ui/button"

export const TabHome = () => {
  const { searchResults, onClearSearch, selectedBook, favorites, addFavorite, removeFavorite } = useBible()

  const [activeTab, setActiveTab] = useState("Inicio")

  useEffect(() => {
    if (selectedBook) {
      setActiveTab("lectura")
    }
  }, [selectedBook])

  useEffect(() => {
    if (searchResults) {
      setActiveTab("lectura")
    }
  }, [searchResults])

  const handleTabChange = (value: string) => {
    setActiveTab(value)
    if (value !== "lectura") {
      onClearSearch()
    }
  }

  return (
    <div className="w-full max-w-4xl mx-auto px-4 sm:px-6 lg:px-8">
      <Tabs value={activeTab} onValueChange={handleTabChange} className="w-full mt-4">
        <TabsList className="w-full flex justify-center mb-6 bg-muted p-1 rounded-lg">
          <TabsTrigger
            value="Inicio"
            className="flex-1 px-3 sm:px-6 py-2 sm:py-3 text-sm font-medium transition-colors duration-100 ease-in-out rounded-md data-[state=active]:bg-slate-900 data-[state=active]:text-slate-50 data-[state=active]:shadow-sm"
          >
            Inicio
          </TabsTrigger>
          <TabsTrigger
            value="lectura"
            className="flex-1 px-3 sm:px-6 py-2 sm:py-3 text-sm font-medium transition-colors duration-100 ease-in-out rounded-md data-[state=active]:bg-slate-900 data-[state=active]:text-slate-50 data-[state=active]:shadow-sm"
          >
            Lectura
          </TabsTrigger>
          <TabsTrigger
            value="favoritos"
            className="relative flex-1 px-3 sm:px-6 py-2 sm:py-3 text-sm font-medium transition-colors duration-100 ease-in-out rounded-md data-[state=active]:bg-slate-900 data-[state=active]:text-slate-50 data-[state=active]:shadow-sm"
          >
            Favoritos
            {favorites.length > 0 && (
              <span className="absolute -top-2 -right-2 bg-red-500 text-white text-xs font-bold px-2 py-1 rounded-full min-w-[1.5rem] flex items-center justify-center">
                {favorites.length}
              </span>
            )}
          </TabsTrigger>
        </TabsList>
        <div className="min-h-[500px] overflow-y-auto">
          <TabsContent
            value="Inicio"
            className="mt-4 p-4 rounded-lg bg-card text-card-foreground shadow-sm transition-opacity duration-300 ease-in-out"
          >
            <Welcome />
          </TabsContent>
          <TabsContent value="lectura">
            <BibleVerse
              searchResults={searchResults}
              onClearSearch={onClearSearch}
              favorites={favorites}
              addFavorite={addFavorite}
              removeFavorite={removeFavorite}
            />
          </TabsContent>
          <TabsContent
            value="favoritos"
            className="mt-4 p-4 rounded-lg bg-card text-card-foreground shadow-sm transition-opacity duration-300 ease-in-out"
          >
            <h2 className="text-2xl font-bold mb-4">Favoritos</h2>

            {favorites.length > 0 ? (
              <div className="grid gap-4 sm:grid-cols-2 lg:grid-cols-3">
                {favorites.map((favorite, index) => (
                  <div key={index} className="bg-white p-4 rounded-lg shadow-md flex flex-col justify-between">
                    <div>
                      <p className="text-sm text-slate-500 font-semibold mb-2">{favorite.reference}</p>
                      <p className="text-base text-slate-800">{favorite.text}</p>
                    </div>
                    <div className="flex justify-between items-center mt-4">
                      <Button
                        variant="ghost"
                        size="icon"
                        onClick={() => removeFavorite(favorite.id)}
                        className="text-red-500 hover:text-red-600 hover:bg-red-100"
                      >
                        <Heart className="w-5 h-5" />
                      </Button>
                      <Button
                        variant="ghost"
                        size="icon"
                        className="text-blue-500 hover:text-blue-600 hover:bg-blue-100"
                      >
                        <Share2 className="w-5 h-5" />
                      </Button>
                    </div>
                  </div>
                ))}
              </div>
            ) : (
              <div className="text-center py-12">
                <Heart className="w-16 h-16 text-slate-300 mx-auto mb-4" />
                <p className="text-xl font-semibold text-slate-600 mb-2">No hay favoritos aún</p>
                <p className="text-slate-500 mb-6">Guarda tus versículos favoritos para acceder a ellos fácilmente</p>
                <Button
                  onClick={() => handleTabChange("lectura")}
                  className="bg-slate-900 text-white hover:bg-slate-800"
                >
                  Ir a Lectura
                </Button>
              </div>
            )}
          </TabsContent>
        </div>
      </Tabs>
    </div>
  )
}

