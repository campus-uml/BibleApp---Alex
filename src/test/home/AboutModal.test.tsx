import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, it, expect } from "vitest";
import "@testing-library/jest-dom";
import { AboutModal } from "../../components/home/AboutModal";

describe("AboutModal", () => {
  it("renderiza el botón del modal", () => {
    const { asFragment } = render(<AboutModal />);
    expect(screen.getByRole("button")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("abre el modal al hacer clic en el botón", async () => {
    const { asFragment } = render(<AboutModal />);
    const button = screen.getByRole("button");
    await userEvent.click(button);
    expect(screen.getByText("Acerca del Desarrollador")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("cerrar el modal al dar clik fuera de el", async () => {
    const { asFragment } = render(<AboutModal />);
    const button = screen.getByRole("button");
    await userEvent.click(button);

    const modalTitle = screen.getByText("Acerca del Desarrollador");
    expect(modalTitle).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
