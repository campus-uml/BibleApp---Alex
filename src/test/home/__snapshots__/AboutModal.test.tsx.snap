// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutModal > abre el modal al hacer clic en el botón 1`] = `
<DocumentFragment>
  <button
    aria-controls="radix-:r3:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 flex items-center gap-2 dark:text-slate-50"
    data-state="open"
    type="button"
  >
    ?
  </button>
</DocumentFragment>
`;

exports[`AboutModal > cerrar el modal al dar clik fuera de el 1`] = `
<DocumentFragment>
  <button
    aria-controls="radix-:r6:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 flex items-center gap-2 dark:text-slate-50"
    data-state="open"
    type="button"
  >
    ?
  </button>
</DocumentFragment>
`;

exports[`AboutModal > renderiza el botón del modal 1`] = `
<DocumentFragment>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 flex items-center gap-2 dark:text-slate-50"
    data-state="closed"
    type="button"
  >
    ?
  </button>
</DocumentFragment>
`;
